official/nlp/modeling/models/electra_pretrainer.py

```diff
@@ -0,0 +1,159 @@
+        'activation': activation,
+        'initializer': initializer,
+        'output': output,
+    }
+
+    # We want to use the inputs of the passed network as the inputs to this
+    # Model. To do this, we need to keep a copy of the network inputs for use
+    # when we construct the Model object at the end of init. (We keep a copy
+    # because we'll be adding another tensor to the copy later.)
+    network_inputs = network.inputs
+    inputs = copy.copy(network_inputs)
+    # Because we have a copy of inputs to create this Model object, we can
+    # invoke the Network object with its own input tensors to start the Model.
+    # Note that, because of how deferred construction happens, we can't use
+    # the copy of the list here - by the time the network is invoked, the list
+    # object contains the additional input added below.
+    sequence_output, cls_output = network(network_inputs)
+
+    sequence_output_length = sequence_output.shape.as_list()[1]
+    if sequence_output_length < num_token_predictions:
+      raise ValueError(
+          "The passed network's output length is %s, which is less than the "
+          'requested num_token_predictions %s.' %
+          (sequence_output_length, num_token_predictions))
+
+    masked_lm_positions = tf.keras.layers.Input(
+        shape=(num_token_predictions,),
+        name='masked_lm_positions',
+        dtype=tf.int32)
+    inputs.append(masked_lm_positions)
+    self.masked_lm = networks.MaskedLM(
+        num_predictions=num_token_predictions,
+        input_width=sequence_output.shape[-1],
+        source_network=network,
+        activation=activation,
+        initializer=initializer,
+        output='logits',
+        name='generator')
+    lm_outputs = tf.identity(self.masked_lm([sequence_output, masked_lm_positions]), "mlm_logits")
+
+    #Generates fake data to train discriminator on
+    fake_data, labels = _get_fake_data(inputs[0], lm_outputs, masked_lm_positions)
+
+    other_output, other_cls_output = discriminator([fake_data, inputs[1], inputs[2]])
+    self.discrimnator = networks.Discriminator(
+        input_width=other_output.shape[-1],
+        source_network=discriminator,
+        activation=activation,
+        initializer=initializer,
+        name='discriminator'
+    )
+
+    discrim_outputs = self.discrimnator(other_output)
+
+    super(ElectraPretrainer, self).__init__(
+        inputs=inputs, outputs=[lm_outputs, discrim_outputs, labels], **kwargs)
+
+  def get_config(self):
+    return self._config
+
+  @classmethod
+  def from_config(cls, config, custom_objects=None):
+    return cls(**config)
+
+
+def _get_fake_data(orig_sent, predictions, maskedlm_ids):
+  tokids = tf.stop_gradient(tf.math.argmax(predictions,axis=-1,output_type=tf.dtypes.int32))
+  updatedids, mask = _scatter_update(orig_sent,maskedlm_ids,tokids)
+  labels = mask*(1-tf.cast(tf.equal(orig_sent,updatedids),tf.int32))
+  return updatedids, labels
```

Comment:
nit: please add spaces after commas, and around operations such as "*", "-". Also please add two empty lines after this function.